### DISCUSSION TITLE
Can you create a multi-line commit?

### DISCUSSION THREAD

**Discussion by cubanx** [jesseduffield/lazygit#1650] [Q&A] [answered]
I figured out you can use the editor, but is there a way to do a multi-line commit?

So we can do a blank line after the first < 50 char line, then enter more details?

I tried Shift-Enter and Ctrl-Enter, but neither worked.

Thanks!

**Comment by cubanx** [comment, marked as answer]
Ok, my bad here, it *is* `Alt-Enter` but my terminal was using `Alt-Enter` for another shortcut

I have this straightened out and we're all good, sorry if anyone wasted any time on this.

**Reply by ShortCipher** [reply]
Thanks! This must be a new-ish feature! I tried to figure out how to do this a year ago and the best I came up with was to use `c` to commit, and then use `R` from the "Commits" panel to reword the commit with an editor.

This works for me in Linux, now if I could just figure out the right keys for macOS.

**Reply by luszczek** [reply]
Indeed, `Alt-Enter` could be hard to "type" with native OS and terminal intervening. Emacs had the same issue for decades and a suggested workaround always was to hit `Esc` instead of `Alt`. I've tried `Esc-Enter` and it worked with `lazygit`using the terminal emulators I use on both Linux and macOS.

**Reply by mark2185** [reply]
@luszczek multiline in the commit's body should work out of the box by pressing `Enter` for the past few versions of `lazygit`.

**Reply by luszczek** [reply]
I think the question was about a different dialog box. One dialog box only allows to enter a single line comment and requires `Alt-Enter` to add multi-line comment. For this dialog box, `Enter` simply accepts the single line commit message and records the commit with Git. The other dialog box has two input areas: one for single line comment and the other for multi-line comment. My version already has both types of dialog box.
